conventional-changelog-lighthouse/index.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+const readFileSync = require('fs').readFileSync;
+const resolve = require('path').resolve;
+const mainTemplate = readFileSync(resolve(__dirname, 'templates/template.hbs')).toString();
+const headerPartial = readFileSync(resolve(__dirname, 'templates/header.hbs')).toString();
+const commitPartial = readFileSync(resolve(__dirname, 'templates/commit.hbs')).toString();
+
+const parserOpts = {
+  headerPattern: /^(\w*)(?:\((.*)\))?: (.*)$/,
+  headerCorrespondence: [
+    'type',
+    'scope',
+    'message',
+  ],
+};
+
+const writerOpts = {
+  mainTemplate,
+  headerPartial,
+  commitPartial,
+  transform: commit => {
+    if (typeof commit.hash === 'string') {
+      commit.hash = commit.hash.substring(0, 7);
+    }
+
+    if (commit.type === 'test') {
+      commit.type = 'tests';
+    }
+
+    if (commit.type) {
+      commit.type = commit.type.replace(/_/g, ' ');
+      commit.type = commit.type.substring(0, 1).toUpperCase() + commit.type.substring(1);
+    } else {
+      commit.type = 'Misc';
+    }
+
+    let pullRequestMatch = commit.header.match(/\(#(\d+)\)/);
```

Comment:
what's the difference between this case and the `message` one below? Is `header` the initial line of the commit message?

Reply:
Also, should we be checking that the PR # is at the end of the string? (the default positioning from the squash and merge button). A #XXXX elsewhere could be e.g. a reference to an issue

Reply:
header is indeed the first line and message is the rest. When I only ran it on message I didn't get all the PR references if I check both I do.

For the regex, i checked the last 50 commits and we never refer an issue in the commit header but better save than sorry so edited the regex!